packages/server/src/filter-test.ts

```diff
@@ -0,0 +1,33 @@
+import { Test } from '@bigtest/suite';
+import * as path from 'path';
+import * as fs from 'fs';
+
+interface Options {
+  files: string[];
+}
+
+function comparePaths(a: string, b: string) {
+  return path.resolve(a) === path.resolve(b);
+}
+
+export function filterTest(test: Test, options: Options): Test {
+  let children: Test[] = [];
+
+  if(options.files.length) {
+    for(let file of options.files) {
+      let child = test.children.find((c) => c.path && comparePaths(c.path, file));
+
+      if(child) {
+        children.push(child);
+      } else if(fs.existsSync(file)) {
+        throw new Error(`file with path '${path.resolve(file)}' exists, but is not part of your test files, if you want to run this test, adjust the 'testFiles' setting in 'bigtest.json'`)
```

Comment:
Might be worth printing out what the test files pattern is so when the error is encountered it can be diagnosed quickly.

Reply:
I'm unsure of what you would like for this to say. Can you suggest a wording for this?

Reply:
I'm mainly thinking of ways I've been bitten by this over the years. One would be trying to run `tests/foos/foo.test.ts`, but the pattern in `bigtest.json` is `tests/*/*.test.ts` that doesn't expand subdirectories. What would have helped in that instance is something like:

```
The file `tests/foos/foo.test.ts` exists, but does not match the "files" pattern 'tests/*/*.test.ts`. If you want to run this test, you will need to adjust the "files" field in bigtest.json.
```

Reply:
Improved error message:

<img width="1027" alt="Screenshot 2020-09-24 at 12 50 12" src="https://user-images.githubusercontent.com/134/94135983-86349880-fe64-11ea-8f9b-2141ea80c814.png">